easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -442,17 +442,18 @@ def copy(self):
 
         return ec
 
-    def update(self, key, value):
+    def update(self, key, value, allow_duplicate=True):
         """
         Update a string configuration value with a value (i.e. append to it).
         """
         prev_value = self[key]
-        if isinstance(prev_value, basestring):
-            self[key] = '%s %s ' % (prev_value, value)
-        elif isinstance(prev_value, list):
-            self[key] = prev_value + value
-        else:
-            raise EasyBuildError("Can't update configuration value for %s, because it's not a string or list.", key)
+        if allow_duplicate or value not in prev_value:
```

Comment:
@mboisson This check isn't entirely correct in the case of `prev_value` (and hence `value`) being a list, since:

```python
>>> [2,3] in [1,2,3]
False
>>> [2,3] not in [1,2,3]
True
```

Reply:
Ah, I hadn't considered that `value` could be a list too, I assumed that we'ld be adding a single element to `prev_value`. I'll fix that.

Reply:
fixed in latest commit